int32 id
int32 lid
float64 width
int8 color
int32 type
int32 linkid
